pubsublite/admin.go

```diff
@@ -144,28 +183,41 @@ func (ac *AdminClient) UpdateSubscription(ctx context.Context, config Subscripti
 	if err != nil {
 		return nil, err
 	}
-	return protoToSubscriptionConfig(subspb)
+	return protoToSubscriptionConfig(subspb), nil
 }
 
-// DeleteSubscription deletes a subscription.
-func (ac *AdminClient) DeleteSubscription(ctx context.Context, subscription SubscriptionPath) error {
-	return ac.admin.DeleteSubscription(ctx, &pb.DeleteSubscriptionRequest{Name: subscription.String()})
+// DeleteSubscription deletes a subscription. A valid subscription path has the
+// format: "projects/PROJECT_ID/locations/ZONE/subscriptions/SUBSCRIPTION_ID".
+func (ac *AdminClient) DeleteSubscription(ctx context.Context, subscription string) error {
+	if _, err := wire.ParseSubscriptionPath(subscription); err != nil {
+		return err
+	}
+	return ac.admin.DeleteSubscription(ctx, &pb.DeleteSubscriptionRequest{Name: subscription})
 }
 
-// Subscription retrieves the configuration of a subscription.
-func (ac *AdminClient) Subscription(ctx context.Context, subscription SubscriptionPath) (*SubscriptionConfig, error) {
-	subspb, err := ac.admin.GetSubscription(ctx, &pb.GetSubscriptionRequest{Name: subscription.String()})
+// Subscription retrieves the configuration of a subscription. A valid
+// subscription name has the format:
+// "projects/PROJECT_ID/locations/ZONE/subscriptions/SUBSCRIPTION_ID".
+func (ac *AdminClient) Subscription(ctx context.Context, subscription string) (*SubscriptionConfig, error) {
+	if _, err := wire.ParseSubscriptionPath(subscription); err != nil {
+		return nil, err
+	}
+	subspb, err := ac.admin.GetSubscription(ctx, &pb.GetSubscriptionRequest{Name: subscription})
 	if err != nil {
 		return nil, err
 	}
-	return protoToSubscriptionConfig(subspb)
+	return protoToSubscriptionConfig(subspb), nil
 }
 
 // Subscriptions retrieves the list of subscription configs for a given project
-// and zone.
-func (ac *AdminClient) Subscriptions(ctx context.Context, location LocationPath) *SubscriptionIterator {
+// and zone. A valid parent path has the format:
+// "projects/PROJECT_ID/locations/ZONE".
+func (ac *AdminClient) Subscriptions(ctx context.Context, parent string) *SubscriptionIterator {
```

Comment:
Is there a reason we can't return (*SubscriptionIterator, err) here instead? That seems better than embedding the error in the iterator itself.

Reply:
It seems to be convention for the list methods to return just the iterator and then fail Next() if the RPC fails. So I just tried to be consistent with the other Go libs and gapic generated libs.